publisher/kafka.go

```diff
@@ -50,7 +50,7 @@ type Kafka struct {
 // ProduceBulk messages to kafka. Block until all messages are sent. Return array of error. Order of Errors is guaranteed.
 // DeliveryChannel needs to be exclusive. DeliveryChannel is exposed for recyclability purpose.
 func (pr *Kafka) ProduceBulk(events []*pb.Event, deliveryChannel chan kafka.Event) error {
-	errors := make([]error, len(events))
+	errors := make([]*EventError, len(events))
```

Comment:
Any reason to use pointers here?

Reply:
This is a pointer because by default the `EventError` struct will not be nil, it will have some default values. So in `worker`, we can't do idiomatic error checks like `if err != nil`, that's why made it a pointer.